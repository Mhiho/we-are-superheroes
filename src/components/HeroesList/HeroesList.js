import React from "react";
import Hero from "../Hero/Hero";

const heroesList = (props) => {
  return (
    <>
      {(props.results || []).map((hero) => {
        return (
          <div className="row w-100">
            <ul
              style={{ cursor: "pointer" }}
              className="list-group w-100 ml-5 mr-5"
            >
              <Hero
                key={hero.id}
                name={hero.name}
                src={hero.image.url}
                click={() => props.passId(hero.id)}
              />
            </ul>
          </div>
        );
      })}
    </>
  );
};

export default heroesList;
