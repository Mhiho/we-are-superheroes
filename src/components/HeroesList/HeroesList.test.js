import React from 'react';
import renderer from 'react-test-renderer';

import HeroesList from './HeroesList';

it('renders correctly when there is no props', () => {
  const tree = renderer.create(<HeroesList />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly where there are props', ()=>{
    const src = 'string'
    const name = 'string'
    const tree = renderer.create(<HeroesList name={name} src={src} />).toJSON();
    expect(tree).toMatchSnapshot();
})
