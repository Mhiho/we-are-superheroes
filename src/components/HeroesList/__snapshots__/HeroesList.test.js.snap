// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is no props 1`] = `null`;

exports[`renders correctly where there are props 1`] = `null`;
